doc/release/upcoming_changes/15118.change.rst

```diff
@@ -0,0 +1,7 @@
+Remove handling of extra argument to ``__array__``
+--------------------------------------------------
+A code path and test have been in the code since NumPy 0.4 for a two-argument
+variant of ``__array__(dtype=None, context=None)``. It was activated when
+calling ``ufunc.reduce(op)`` if ``op.__array__`` existed. However that variant
```

Comment:
This doesn't sound correct - the test you removed does not appear to call `reduce`

Reply:
I think there is another simililar thing in the argument parsing used by normal ufunc calls (currently inside `PyUfunc_GenericFunction`.

It may be good to add an `assert` for context being NULL if we never pass it on?

Reply:
Calling ` ncu.maximum(np.zeros(1), a)` hits a breakpoint in `PyUFunc_GenericReduction(ufunc, args, ...)` with the `maximum` ufunc

Reply:
`ncu.maximum` is just `np.maximum`, right?

Reply:
```
>>> np.core.umath.maximum is np.maximum
True
```

It's misleading. The reduction is called via `array_repr`, when printing the result. The ufunc called `__array__` via `PyUfunc_GenericFunction` (well, `get_ufunc_arguments`) as @seberg pointed out.

Fixing the release note